src/scheduler.rs

```diff
@@ -434,6 +434,82 @@ impl Scheduler {
         }
         Ok(())
     }
+
+    /// Send a PayJoin from LND using automatic coin selection and
+    /// automatic fee rate of 2 target confirmations.
+    ///
+    /// `danger_accept_self_signed_uri` should be used for testing only.
+    /// PayJoins are vulnerable to man-in-the-middle attacks, so they must
+    /// communicate over secure connections in production.
+    pub async fn send_payjoin<'a>(
+        &self,
+        uri: bip78::Uri<'_>,
+        danger_accept_self_signed_uri: bool,
+    ) -> Result<Txid, SchedulerError> {
+        log::debug!("get original_psbt");
+        let pj_uri = bip78::UriExt::check_pj_supported(uri)
+            .map_err(|e| SchedulerError::UriDoesNotSupportPayJoin(e.to_string()))?;
+        let (original_psbt, leased_utxos) = if let Some(amount) = pj_uri.amount {
+            log::debug!("funding original_psbt");
+            self.lnd.fund_original_psbt(&pj_uri.address, amount).await?
+        } else {
+            return Err(SchedulerError::UriDoesNotSupportPayJoin(
+                "Missing amount. Make sure the bip21 uri specifies an amount".to_string(),
+            ));
+        };
+        log::debug!("sign original_psbt");
+        let original_psbt = self.lnd.sign_psbt(original_psbt).await?;
+        log::debug!("request_payjoin");
+        let res = self.request_payjoin(pj_uri, original_psbt, danger_accept_self_signed_uri).await;
+        if res.is_err() {
+            log::debug!("release_utxos");
+            self.lnd.release_utxos(leased_utxos).await?;
+        } // else, those utxos are now spent and don't need to be released
+        res
+    }
+
+    async fn request_payjoin(
+        &self,
+        pj_uri: bip78::PjUri<'_>,
+        original_psbt: PartiallySignedTransaction,
+        danger_accept_self_signed_uri: bool,
+    ) -> Result<Txid, SchedulerError> {
+        use bip78::PjUriExt;
+
+        let pj_params = bip78::sender::Configuration::with_fee_contribution(
+            bip78::bitcoin::Amount::from_sat(10000),
```

Comment:
So this is our "maxadditionalfeecontribution", seems reasonable 👍

Reply:
I think you caught a bug here actually. This should be changed to `Configuration::non_incentivizing()` or else this might send more sats than intended. With_fee_contribution could be supported later.